### PR TITLE
Update gradle to 6.7.1 version.

### PR DESCRIPTION
**Pull request by Tapchicoma**
Set minimal supported Gradle version to 6.0 and check it on plugin apply.

### CODE REVIEW

File: plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/KtlintBasePlugin.kt [hunk 1]
```diff
@@ -23,16 +22,6 @@ open class KtlintBasePlugin : Plugin<Project> {
...
-        target.tasks.withType(BaseKtlintCheckTask::class.java).configureEach {
-            val objects = target.objects
-            val gradleVersion = GradleVersion.version(target.gradle.gradleVersion)
-            if (gradleVersion < GradleVersion.version("6.0")) {
-                it.runner = objects.newInstance(JavaExecKtLintRunner::class.java, target)
-            } else {
-                it.runner = objects.newInstance(WorkerApiKtLintRunner::class.java)
-            }
-        }
-
```
Comment (JLLeitschuh) [inline plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/KtlintBasePlugin.kt:35]: What happened to this?

Comment (Tapchicoma) [reply plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/KtlintBasePlugin.kt:35]: `JavaExecKtLintRunner` was used when plugin was running with Gradle prior to 6.0 version. As minimal supported Gradle version was bumped to `6.5.1` - I've removed it.

Comment (JLLeitschuh) [reply plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/KtlintBasePlugin.kt:35]: Ah, moved up here:

https://github.com/JLLeitschuh/ktlint-gradle/pull/420/files#diff-16dd0c2785c1b92755de7e2eaef5f722a769733f7917e1dda9995c5d60f17123R159

Makes sense to me! Thanks!

File: CHANGELOG.md [hunk 2]
```diff
@@ -18,6 +18,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/).
 
     **Breaking** - removed build variants meta tasks. Minimum supported AGP version is 4.0.0.
   - Updated shadow plugin to 6.1.0 version.
+  - Updated Gradle to `6.7.1` version
+  - Set minimal supported Gradle version to `6.5.1`
```
Comment (JLLeitschuh) [inline CHANGELOG.md:22]: Seems like a pretty restricted version range to support. Didn't we want to support to back to 6.0?

Comment (Tapchicoma) [reply CHANGELOG.md]: Updated commit to support it back to 6.0: cde95a1

**Review by JLLeitschuh — APPROVED** [review]
LGTM!

**Pull request merged**